src/plugins/data/public/ui/query_editor/query_editor_top_row.tsx

```diff
@@ -323,6 +330,8 @@ export default function QueryEditorTopRow(props: QueryEditorTopRowProps) {
       direction="column"
       justifyContent="flexEnd"
     >
+      {props?.datasetSelectorRef?.current &&
+        createPortal(datasetSelector, props.datasetSelectorRef.current)}
```

Comment:
attempting to keep all the render logic relatively tight together since we got it pretty nailed down a couple weeeks ago. otherwise we will have to pass stateful props which felt like too new dev vs keeping it in the data plugin and portalling it where we want it.